app/models/manageiq/providers/vmware/cloud_manager/event_catcher.rb

```diff
@@ -8,4 +8,14 @@ def self.ems_class
   def self.settings_name
     :event_catcher_vmware_cloud
   end
+
+  def self.all_valid_ems_in_zone
+    # Only run event catcher for those vClouds that have credentials for it.
+    # NOTE: ATM it's safest to check if hostname is non-empty string because
+    # frontend seems to be inserting empty Authentication and Endpoint even
+    # if user opts-in for "None" in AMQP tab.
+    super.select do |ems|
+      ems.endpoints.detect { |e| e.role == 'amqp' && e.hostname.present? }
```

Comment:
I think `ems.endpoints.any?` would be more obvious since you aren't trying to actually return the endpoint here, just treat its existence as a boolean

Reply:
The man is right! Thanks.